ansible_base/authentication/utils/authentication.py

```diff
@@ -95,13 +95,21 @@ def check_system_username(uid: str) -> None:
 
 
 def determine_username_from_uid_social(**kwargs) -> dict:
-    uid_field = getattr(kwargs.get('backend', None), 'ID_KEY', 'username')
-    if uid_field is None:
-        uid_field = 'username'
+    backend = kwargs.get('backend', None)
+    uid_field = 'username'
+    if backend:
+        # Update our fallback if the authenticator has an ID_KEY field
+        if hasattr(backend, "ID_KEY"):
```

Comment:
```suggestion
        # Update our fallback if the authenticator has an ID_KEY field, ignore if ID_KEY is none
        if getattr(backend, "ID_KEY", None) is not None:
```
This is probably something we want to do.